drivers/dma/dma_mcux_lpc.c

```diff
@@ -179,12 +182,53 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if ( config->source_chaining_en || config->dest_chaining_en) {
+		if ( !k_sem_take(&is_trig_mux_in_use, K_MSEC(200))) {
```

Comment:
This might be a niche thing, but I prefer my approach.
In my opinion it's more readable and control flow is literally in line.
This also helps the compiler and the programmer understand the scope 
of any section of code.

Reply:
No, nested conditionals are much harder to read

Reply:
okay well at least make the timeout immediate because there is not really expected to be any reason to wait